optaplanner-benchmark/src/test/java/org/optaplanner/benchmark/config/PlannerBenchmarkConfigTest.java

```diff
@@ -193,57 +167,21 @@ public void calculateWarmUpTimeMillisSpentLimit() {
 
     @Test
     public void xmlConfigFileRemainsSameAfterReadWrite() throws IOException {
-        String benchmarkConfigResource = "org/optaplanner/benchmark/config/testdataBenchmarkConfigNoInheritence.xml";
-        String originalXml = IOUtils.toString(
-                getClass().getClassLoader().getResourceAsStream(benchmarkConfigResource), StandardCharsets.UTF_8);
-        PlannerBenchmarkConfig benchmarkConfig = PlannerBenchmarkConfig.createFromXmlResource(benchmarkConfigResource);
-        assertThat(PlannerBenchmarkFactory.create(benchmarkConfig).buildPlannerBenchmark(new TestdataSolution())).isNotNull();
-        XStream xStream = XStreamConfigReader.buildXStreamPortable(getClass().getClassLoader(), PlannerBenchmarkConfig.class);
-        xStream.setMode(XStream.NO_REFERENCES);
-        String savedXml = xStream.toXML(benchmarkConfig);
-        assertThat(savedXml.trim()).isEqualTo(originalXml.trim());
-    }
+        JaxbIO<PlannerBenchmarkConfig> xmlIO = new JaxbIO<>(PlannerBenchmarkConfig.class);
+        PlannerBenchmarkConfig jaxbBenchmarkConfig;
 
-    @Test
-    public void jaxbXmlConfigFileRemainsSameAfterReadWrite() throws IOException {
-        PlannerBenchmarkConfig jaxbBenchmarkConfig = unmarshallBenchmarkConfigFromResource(TEST_PLANNER_BENCHMARK_CONFIG);
+        try (Reader reader =
+                new InputStreamReader(PlannerBenchmarkConfigTest.class.getResourceAsStream(TEST_PLANNER_BENCHMARK_CONFIG))) {
+            jaxbBenchmarkConfig = xmlIO.read(reader);
+        }
 
         Writer stringWriter = new StringWriter();
-        marshall(jaxbBenchmarkConfig, stringWriter);
+        xmlIO.write(jaxbBenchmarkConfig, stringWriter);
         String jaxbString = stringWriter.toString();
 
         String originalXml = IOUtils.toString(
                 PlannerBenchmarkConfigTest.class.getResourceAsStream(TEST_PLANNER_BENCHMARK_CONFIG), StandardCharsets.UTF_8);
 
         assertThat(jaxbString.trim()).isEqualToNormalizingNewlines(originalXml.trim());
```

Comment:
Do we want to normalize the newlines?

The XML in the benchmark report (so they can easily copy paste it from there into a solverConfig.xml) should contain linux line endings, even on windows.
Do we want the xmlIO to always use linux line ending, regardless of the platform?

Reply:
I will look into this for the BenchmarkReport, together with portable file paths, which is already something on my list to fix. Are you fine with me addressing it in a separate PR?

Reply:
https://issues.redhat.com/browse/PLANNER-2062